ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -174,18 +173,42 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		}
 		return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
 	case rayv1.JobDeploymentStatusRunning:
+		if shouldUpdate, err := r.updateStatusToSuspendingIfNeeded(ctx, rayJobInstance); shouldUpdate {
```

Comment:
Without this PR, to transition the status to `Suspending`, we need to submit a `GetJobInfo` request and update the status twice, using both `updateState` and `updateStatusToSuspendingIfNeeded`. With this PR, we just need to update the status once to transition the status to `Suspending`.